### PR TITLE
BFD-1568 Introduce bfd-ansible container image

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
[BFD-1568](https://jira.cms.gov/browse/BFD-1568)

**User Story or Bug Summary:**

Introduce bfd-ansible container image
---

### What Does This PR Do?

This PR introduces a Dockerfile and CI workflow that builds a base container image, `bfd-ansible`, similar to the images that are currently built on-demand as part of the `ci-ansible.yml` workflow, and stores it in the CMSgov artifact repository/container registry–[see the non-trunk builds here](https://github.com/CMSgov/beneficiary-fhir-data/pkgs/container/bfd-ansible).

There are no updates to existing functionality as part of this request and newly-introduced artifacts are not yet in use. The CI here only operates on the protected `master` branch when there are changes to one or both of `.github/workflows/ci-base-container-image.yml` or `ops/Dockerfile`. The workflow steps themselves eschew third-party github actions, e.g. [login-action](https://github.com/docker/login-action), [build-push-action](https://github.com/docker/build-push-action) in favor of short, more portable (and more obvious) shell scripts.

#### ops/Dockerfile 

Although `ops/Dockerfile` is a _new_ file, I'm attempting to keep a light touch here by largely using existing Dockerfiles from the test harnesses found in the ansible role directories, i.e. [bfd-pipeline](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/ops/ansible/roles/bfd-pipeline/test/docker_platforms/centos_7/Dockerfile) and [bfd-server](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/ops/ansible/roles/bfd-server/test/docker_platforms/centos_7/Dockerfile). There are some _small_ innovations:
* parmaterized upstream container image for future use
* introducing the LABEL directive to associate this container image as approrpriate
* introduced some additional dependencies that consistently installed in the prerequisite setup portion of the tests, `unzip` and `selinux-policy`.
* overall flattened image layers

#### .github/workflows/ci-base-container-image.yml
This is all fairly straight forward. There may be other controversial things here, but I think the image tagging strategy could be among them. I've elected to tag on the branch name as well as the specific short commit sha. I'm assuming the frequency of change should be rather low, so just the branch name, `master` or even the anti-pattern `latest` would probably be fine.

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+         -exec rm \{} \;
+RUN systemctl set-default multi-user.target
+STOPSIGNAL SIGRTMIN+3
+
+# Install and start SSH.
+#
+# References:
+#
+# * <https://stackoverflow.com/questions/18173889/cannot-access-centos-sshd-on-docker>
+RUN yum update -y
```
[the remark this answers is not in the document]
Comment (keithdadkins) [reply ops/Dockerfile:29]: Is selinux even enabled?

File: .github/workflows/ci-base-container-image.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    branches:
+      - master
+    paths:
+      - ops/Dockerfile
+      - .github/workflows/ci-base-container-image.yml
+
+jobs:
+  build-container:
+    name: Build and Deliver Container
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (mjburling) [reply .github/workflows/ci-base-container-image.yml:13]: It's not available:
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

This is only the dockerhost, after all.

File: ops/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
+ARG UPSTREAM_IMAGE="centos:7"
```
[the remark this answers is not in the document]
Comment (mjburling) [reply ops/Dockerfile:1]: It's just what's already there for the ansible portions. I've captured this in [BFD-1571](https://jira.cms.gov/browse/BFD-1571).

File: ops/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjburling) [reply ops/Dockerfile:29]: It's not enabled, but we do install this package in the as part of the [bfd-server role's test harness](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/ops/ansible/roles/bfd-server/test/test_basic.yml#L32).

**Review by keithdadkins — DISMISSED** [review]

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**